sentry-ruby/lib/sentry/client.rb

```diff
@@ -111,6 +111,7 @@ def event_from_transaction(transaction)
         event.timestamp = transaction.timestamp
         event.start_timestamp = transaction.start_timestamp
         event.tags = transaction.tags
+        event.dynamic_sampling_context = transaction.baggage&.dynamic_sampling_context
```

Comment:
Now that we have more transaction related attributes being passed to the event, do you think we can pass the transaction as an argument `TransactionEvent.new(configuration: configuration, transaction: transaction)` and encapsulate these stuff inside `TransactionEvent`'s constructor?

The below 2 lines can be moved there too.

Reply:
done